### PR TITLE
Prepare css variables to use cds instead of clr

### PR DESCRIPTION
**Pull request by antgamdia**
### Description of the change

This PR changes some css variables from `clr` to `cds`. It also fixes some inconsistencies in the fallback used colors.

### Benefits

This PR paves the way for the upgrading process from clarity v4 to v5. 

### Possible drawbacks

This PR (and some of the subsequent ones) are aimed at allowing dark mode in clarity v5. Until every PR becomes merged, some css inconsistencies might occur.

### Applicable issues

  - related #2372

### Additional information

The whole set of PRs includes:

- Remove hardcoded css, delegate colors in the cds variables (which is more customizable through themes), minor css fixes.
- Bump up dependencies. Minor fixes.
  - Update modals
  - Update layout
  - Add media query to detect dark mode (for cds)
  - Add dynamic dark mode for clr

### CODE REVIEW

File: dashboard/src/index.scss [hunk 1]
```diff
@@ -1,38 +1,44 @@
...
-  min-width: 0;
-}
-
 svg.icon {
   /* Fix positioning */
   position: relative;
   bottom: -0.125em;
 }
 
-/* Override Hex underline text-decoration in buttons in alerts
```
Comment (antgamdia) [inline dashboard/src/index.scss:18]: Unused css selectors

File: dashboard/src/index.scss [hunk 2]
```diff
@@ -1,38 +1,44 @@
...
+  border: 1px solid var(--cds-alias-object-border-color, #ccc);
+  background-color: var(--cds-alias-object-interaction-background), #fff;
   overflow: auto;
   border-radius: 4px;
   outline: none;
   padding: 20px;
   margin-right: -50%;
   transform: translate(-50%, -50%);
+
+  // Copied from the CdsButton style
```
Comment (antgamdia) [inline dashboard/src/index.scss:27]: this is a workaround for an unlikely modal to match the rest of the styles:
image

File: dashboard/src/components/DeploymentFormBody/BasicDeploymentForm/BasicDeploymentForm.scss [hunk 3]
```diff
@@ -43,7 +43,7 @@
 }
 
 .param-separator {
-  border: 1px solid #f1f1f1;
+  border: 1px solid var(#f1f1f1);
```
Comment (antgamdia) [inline dashboard/src/components/DeploymentFormBody/BasicDeploymentForm/BasicDeploymentForm.scss:46]: Ups, forgot about this one

File: dashboard/src/components/Header/Menu.scss [hunk 4]
```diff
@@ -105,8 +105,14 @@
...
-    color: #fff;
+  a {
+    color: var(--cds-global-color-blue-400, #2ec0ff) !important;
+  }
+  &:visited {
+    color: var(--cds-global-color-lavender-300 #8b8dda) !important;
+  }
+  &:hover {
+    color: var(--cds-global-color-blue-500, #fff) !important;
   }
```
Comment (andresmgot) [inline dashboard/src/components/Header/Menu.scss:116]: why this changes? (we should avoid using !important if possible).

File: dashboard/src/index.scss (hunk 2 above)
Comment (andresmgot) [reply dashboard/src/index.scss:27]: what's the behavior without this `.button`?

**Review by andresmgot — APPROVED** [review]
thanks for reviewing this!

File: dashboard/src/index.scss (hunk 2 above)
Comment (antgamdia) [reply dashboard/src/index.scss:27]: Something like: 
image

File: dashboard/src/components/Header/Menu.scss (hunk 4 above)
Comment (antgamdia) [reply dashboard/src/components/Header/Menu.scss:116]: The `a:link` in the clr-ui css is taking more priority than the css style defined at Menu.scss, so, if not adding the `!important` the bad color won't get overidden.
Pehaps I'm missing something here :S

image

Comment (andresmgot) [reply dashboard/src/components/Header/Menu.scss:116]: you have the answer :) if you add the `:link` modifier to the line `108`, you don't need the `!important` rule

File: dashboard/src/index.scss (hunk 2 above)
Comment (andresmgot) [reply dashboard/src/index.scss:27]: hum, weird, we should be missing something here but I am not sure what

File: dashboard/src/components/Header/Menu.scss (hunk 4 above)
Comment (antgamdia) [reply dashboard/src/components/Header/Menu.scss]: Thanks, after some changes we can get rid of the `!important` modifiers :)

File: dashboard/src/index.scss (hunk 2 above)
Comment (antgamdia) [reply dashboard/src/index.scss]: image

I've just updated this modal to use the cds button as other components do
